server/api/crud/projects.py

```diff
@@ -417,7 +443,24 @@ async def _calculate_pipelines_counters(
                         not in mlrun.run.RunStatuses.stable_statuses()
                     ):
                         project_to_running_pipelines_count[pipeline["project"]] += 1
-
+                    elif "finished_at" in pipeline:
+                        finished_at = datetime.datetime.strptime(
+                            pipeline["finished_at"], "%Y-%m-%d %H:%M:%S%z"
+                        )
+                        if finished_at > datetime.datetime.now().astimezone(
+                            tz=datetime.timezone.utc
+                        ) - datetime.timedelta(days=1):
```

Comment:
We need to be performance-minded, and this query is extremely inefficient, looking for all pipelines from all projects, just so we can filter pipelines from last day here.
For our purposes this should be done with 2 filtered-by-kfp `list_pipelines` - one looking for running pipelines (by filtering on status not being a stable status) and another looking for pipelines completed in the last day. Should check to see if kfp allows such a filter (not all fields can be used to filter) - if it does, then this query should definitely be filtered already in the kfp side.

Reply:
Hey, agree with you but our version doesn't support filtering by `finished_at`.
They added it only in this pr:
https://github.com/kubeflow/pipelines/pull/8662

Reply:
Got it. Thanks for checking.